src/helpers/JpegXL.cpp

```diff
@@ -0,0 +1,125 @@
+    }
+
+    JxlDecoderSetInput(dec.get(), reinterpret_cast<const uint8_t*>(imageRawData), fileInfo.st_size);
+    JxlDecoderCloseInput(dec.get());
+    if (JXL_DEC_BASIC_INFO != JxlDecoderProcessInput(dec.get())) {
+        Debug::log(ERR, "createSurfaceFromJXL: JxlDecoderProcessInput failed");
+        free(imageRawData);
+        exit(1);
+    }
+
+    JxlBasicInfo basic_info;
+    if (JXL_DEC_SUCCESS != JxlDecoderGetBasicInfo(dec.get(), &basic_info)) {
+        Debug::log(ERR, "createSurfaceFromJXL: JxlDecoderGetBasicInfo failed");
+        free(imageRawData);
+        exit(1);
+    }
+
+    auto cairoSurface = cairo_image_surface_create(CAIRO_FORMAT_ARGB32, basic_info.xsize, basic_info.ysize);
+    if (cairo_surface_status(cairoSurface) != CAIRO_STATUS_SUCCESS) {
+        Debug::log(ERR, "createSurfaceFromJXL: Cairo Failed (?)");
+        free(imageRawData);
+        cairo_surface_destroy(cairoSurface);
+        exit(1);
+    }
+
+    const auto     CAIRODATA = cairo_image_surface_get_data(cairoSurface);
+
+    JxlPixelFormat format = {
+        .num_channels = 4,
+        .data_type    = JXL_TYPE_UINT8,
+        .endianness   = JXL_NATIVE_ENDIAN,
+        .align        = 0,
+    };
+
+    for (;;) {
+        JxlDecoderStatus status = JxlDecoderProcessInput(dec.get());
+        if (status == JXL_DEC_ERROR) {
+            Debug::log(ERR, "createSurfaceFromJXL: JxlDecoderProcessInput failed");
+            free(imageRawData);
+            cairo_surface_destroy(cairoSurface);
+            exit(1);
+        } else if (status == JXL_DEC_NEED_MORE_INPUT) {
+            Debug::log(ERR, "createSurfaceFromJXL: JxlDecoderProcessInput expected more input");
+            free(imageRawData);
+            free(cairoSurface);
+            exit(1);
+        } else if (status == JXL_DEC_NEED_IMAGE_OUT_BUFFER) {
+            JxlResizableParallelRunnerSetThreads(runner.get(), JxlResizableParallelRunnerSuggestThreads(basic_info.xsize, basic_info.ysize));
+            size_t buffer_size;
+            if (JXL_DEC_SUCCESS != JxlDecoderImageOutBufferSize(dec.get(), &format, &buffer_size)) {
+                Debug::log(ERR, "createSurfaceFromJXL: JxlDecoderImageOutBufferSize failed");
+                free(imageRawData);
+                cairo_surface_destroy(cairoSurface);
+                exit(1);
+            }
+            if (buffer_size != basic_info.xsize * basic_info.ysize * format.num_channels) {
+                Debug::log(ERR, "createSurfaceFromJXL: invalid output buffer size");
+                free(imageRawData);
+                cairo_surface_destroy(cairoSurface);
+                exit(1);
+            }
+            if (JXL_DEC_SUCCESS != JxlDecoderSetImageOutBuffer(dec.get(), &format, CAIRODATA, buffer_size)) {
+                Debug::log(ERR, "createSurfaceFromJXL: JxlDecoderSetImageOutBuffer failed");
+                free(imageRawData);
+                cairo_surface_destroy(cairoSurface);
+                exit(1);
+            }
```

Comment:
I even tried this with no luck... nothing changed.
```suggestion
            for (size_t i = 0; i < buffer_size; i += format.num_channels) {
                std::swap(CAIRODATA[i], CAIRODATA[i + 2]);
            }
```